src/net_processing.cpp

```diff
@@ -1849,6 +1849,10 @@ void PeerManagerImpl::BlockChecked(const CBlock& block, const BlockValidationSta
     // 3. This is currently the best block we're aware of. We haven't updated
     //    the tip yet so we have no way to check this directly here. Instead we
     //    just check that there are currently no other blocks in flight.
+    //
+    // No need to consider the background validation chainstate's IBD status
+    // (if applicable) because any active chainstate not in IBD will allow
+    // us to judge block validity.
```

Comment:
I think `doc/design/assumeutxo.md` could be updated with the description of the changes implemented here. Beyond, such documentation would constitute a good reference to review the changes themselves and assert their correctness.